ComputerInterface/Views/GameSettings/JoinRoomView.cs

```diff
@@ -16,23 +12,25 @@ namespace ComputerInterface.Views.GameSettings
     public class JoinRoomView : ComputerView
     {
         private readonly UITextInputHandler _textInputHandler;
-
+        private GameObject callbacks;
         private string _joinedRoom;
 
         public JoinRoomView()
         {
             _textInputHandler = new UITextInputHandler();
-
-            GameObject callbacks = new GameObject();
-            callbacks.name = "RoomCallbacks";
-            GameObject.Instantiate(callbacks);
-            JoinRoomViewCallbacks calllbacksComponent = callbacks.AddComponent<JoinRoomViewCallbacks>();
-            calllbacksComponent.view = this;
         }
 
         public override void OnShow(object[] args)
         {
             base.OnShow(args);
+
+            callbacks = new GameObject();
+            callbacks.name = "RoomCallbacks";
+            UnityEngine.Object.DontDestroyOnLoad(callbacks);
+
+            JoinRoomViewCallbacks calllbacksComponent = callbacks.AddComponent<JoinRoomViewCallbacks>();
+            calllbacksComponent.view = this;
+

```

Comment:
Why did you move this? You shouldn't need to create a new GameObject every time

Reply:
Not 100% but I think it was causing some issues?